pytensor/tensor/rewriting/subtensor.py

```diff
@@ -364,21 +370,63 @@ def local_useless_slice(fgraph, node):
             last_useless_slice -= 1
         else:
             break
+
+    new_idxs = list(idxs)[:last_useless_slice]
```

Comment:
I think you can flip the logic and keep track of last useful index(or slice) in your loop below and get rid of the loop above 

So a single pass will do both of the things